iroh-bytes/src/downloader/progress.rs

```diff
@@ -0,0 +1,262 @@
+use std::{
+    collections::{hash_map, HashMap},
+    sync::{
+        atomic::{AtomicU64, Ordering},
+        Arc,
+    },
+};
+
+use anyhow::anyhow;
+use parking_lot::Mutex;
+
+use crate::{
+    export::ExportProgress,
+    get::{db::DownloadProgress, progress::TransferState},
+    util::progress::{FlumeProgressSender, IdGenerator, ProgressSendError, ProgressSender},
+};
+
+use super::DownloadKind;
+
+/// The channel that can be used to subscribe to progress updates.
+pub type ProgressSubscriber = FlumeProgressSender<DownloadProgress>;
```

Comment:
Once it is merged we can make this a BoxedProgressSender so we don't have flume in here...